cassandra-sink/src/test/java/org/springframework/cloud/stream/module/cassandra/sink/CassandraSinkIntegrationTests.java

```diff
@@ -0,0 +1,204 @@
+		EmbeddedCassandraServerHelper.startEmbeddedCassandra(CASSANDRA_STORAGE_CONFIG, "target/embeddedCassandra");
+		cluster = Cluster.builder()
+				.addContactPoint("localhost")
+				.withPort(PORT)
+				.build();
+
+		cluster.connect().execute(String.format("CREATE KEYSPACE IF NOT EXISTS %s" +
+				"  WITH REPLICATION = { 'class' : 'SimpleStrategy', 'replication_factor' : 1 };", CASSANDRA_KEYSPACE));
+
+		cassandraTemplate = new CassandraTemplate(cluster.connect(CASSANDRA_KEYSPACE));
+
+	}
+
+	@AfterClass
+	public static void cleanup() {
+		if (cluster != null) {
+			cluster.close();
+		}
+		EmbeddedCassandraServerHelper.cleanEmbeddedCassandra();
+	}
+
+	@WebIntegrationTest({"spring.cassandra.port=" + PORT,
+		"spring.cassandra.keyspace=" + CASSANDRA_KEYSPACE,
+		"spring.cassandra.entity-base-packages=org.springframework.cloud.stream.module.cassandra.test.domain",
+		"query-type=INSERT"})
+	@DirtiesContext
+	public static class CassandraEntityInsertTests extends CassandraSinkIntegrationTests {
+
+		@Test
+		public void testInsert() throws InterruptedException {
+			Book book = new Book();
+			book.setIsbn(UUIDs.timeBased());
+			book.setTitle("Spring Integration Cassandra");
+			book.setAuthor("Cassandra Guru");
+			book.setPages(521);
+			book.setSaleDate(new Date());
+			book.setInStock(true);
+	
+			sink.input().send(new GenericMessage<>(book));
+	
+			final Select select = QueryBuilder.select().all().from("book");
+	
+			assertEqualsEventually(1, new Supplier<Integer>() {
+	
+				@Override
+				public Integer get() {
+					return cassandraTemplate.select(select, Book.class).size();
+				}
+	
+			});
+	
+			cassandraTemplate.delete(book);
+		}
+	}
+
+
+	@WebIntegrationTest({"spring.cassandra.port=" + PORT,
+		"spring.cassandra.keyspace=" + CASSANDRA_KEYSPACE,
+		"spring.cassandra.init-script=init-db.cql",
+		"query-type=INSERT",
+		"ingest-query=insert into book (isbn, title, author, pages, saleDate, inStock) values (?, ?, ?, ?, ?, ?)"})
+	@DirtiesContext
+	public static class CassandraSinkIngestTests extends CassandraSinkIntegrationTests {
+
+		@Test
+		public void testIngestQuery() throws Exception {
+			List<Book> books = getBookList(5);
+			ObjectMapper objectMapper = new ObjectMapper();
+			objectMapper.configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false);
+			Jackson2JsonObjectMapper mapper = new Jackson2JsonObjectMapper(objectMapper);
+
+			sink.input().send(new GenericMessage<>(mapper.toJson(books)));
+
+			final Select select = QueryBuilder.select().all().from("book");
+
+			assertEqualsEventually(5, new Supplier<Integer>() {
+
+				@Override
+				public Integer get() {
+					return cassandraTemplate.select(select, Book.class).size();
+				}
+
+			});
+
+			cassandraTemplate.truncate("book");
+		}
+	}
+
+	
+	private static List<Book> getBookList(int numBooks) {
+
+		List<Book> books = new ArrayList<>();
+
+		Book b;
+		for (int i = 0; i < numBooks; i++) {
+			b = new Book();
+			b.setIsbn(UUID.randomUUID());
```

Comment:
What is interesting that this test data is the same like for the `insert` test.
From other side we have the table script:
```
isbn    	uuid PRIMARY KEY,
```
not `timeuuid`, like it barks to us for the `insert` test...